File: MATTHEW/leetcode/AIcoding/4.GPT-4o/javascript_solutions/551-student-attendance-record-i.js
var checkRecord = function(s) {
    let aCount = 0;
    let lCount = 0;

    for (let i = 0; i < s.length; i++) {
        if (s[i] === 'A') {
            aCount++;
        }
        if (s[i] === 'L') {
            lCount++;
            if (lCount > 2) return false;
        } else {
            lCount = 0;
        }
    }

    return aCount < 2;
};

File: leetcode/AIcoding/4.GPT-4o/java_solutions/551-student-attendance-record-i.java
public class Solution {
    public boolean checkRecord(String s) {
        int aCount = 0;
        int lCount = 0;

        for (char c : s.toCharArray()) {
            if (c == 'A') {
                aCount++;
                if (aCount > 1) return false;
            }
            if (c == 'L') {
                lCount++;
                if (lCount > 2) return false;
            } else {
                lCount = 0;
            }
        }
        return true;
    }
}